app/src/main/java/com/nutomic/syncthingandroid/activities/MainActivity.java

```diff
@@ -218,20 +218,25 @@ public void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_main);
         mDrawerLayout = findViewById(R.id.drawer_layout);
 
-        mViewPager = findViewById(R.id.pager);
-        mViewPager.setAdapter(mSectionsPagerAdapter);
-
-        TabLayout tabLayout = findViewById(R.id.tabContainer);
-        tabLayout.setupWithViewPager(mViewPager);
-
+        FragmentManager fm = getSupportFragmentManager();
         if (savedInstanceState != null) {
-            FragmentManager fm = getSupportFragmentManager();
             mFolderListFragment = (FolderListFragment) fm.getFragment(
                     savedInstanceState, FolderListFragment.class.getName());
             mDeviceListFragment = (DeviceListFragment) fm.getFragment(
                     savedInstanceState, DeviceListFragment.class.getName());
             mDrawerFragment = (DrawerFragment) fm.getFragment(
                     savedInstanceState, DrawerFragment.class.getName());
+        } else {
+            mFolderListFragment = new FolderListFragment();
+            mDeviceListFragment = new DeviceListFragment();
+            mDrawerFragment = new DrawerFragment();
+        }
+
+        mViewPager = findViewById(R.id.pager);
+        mViewPager.setAdapter(mSectionsPagerAdapter);
```

Comment:
Note: Here is the line after which I expect the android OS' FragmentManager can and will callback to mSectionsPagerAdapter#getItem to retrieve the fragments guaranteed to be non-null at this point of execution.